nixos/modules/services/web-apps/fusion.nix

```diff
@@ -0,0 +1,144 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.fusion;
+in
+{
+  options.services.fusion = {
+    enable = lib.mkEnableOption "Fusion, a lightweight, self-hosted friendly RSS aggregator and reader";
+
+    host = lib.mkOption {
+      type = lib.types.str;
+      default = "0.0.0.0";
+      example = "[::1]";
+      description = "The address that Fusion should listen on.";
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 8080;
+      description = "The port that Fusion should listen on.";
+    };
+
+    passwordFile = lib.mkOption {
+      type = lib.types.path;
+      example = "/run/keys/fusion-password";
+      description = ''
+        A file containing the password for the Fusion web interface.
+
+        This should be a path pointing to a file within a secure directory and NEVER
+        in the Nix store (which is world-readable)!
+      '';
+    };
+
+    tls = lib.mkOption {
+      type = lib.types.nullOr (
+        lib.types.submodule {
+          options = {
+            cert = lib.mkOption {
+              type = lib.types.path;
+              description = "Path to TLS certificate";
+            };
+            key = lib.mkOption {
+              type = lib.types.path;
+              description = "Path to TLS key";
+            };
+          };
+        }
+      );
+      default = null;
+      description = ''
+        The paths to the TLS certificate and key files for Fusion.
+
+        If these options are set, then Fusion can only be accessed through a secure
+        TLS connection. If you are using a reverse proxy like Nginx to handle HTTPS,
+        please leave these unset.
+      '';
+    };
```

Comment:
Not sure if we just want to use nginx here all the time or wire this into acme directly.

Reply:
We're not using Nginx? I'm just saying that if you are using a reverse proxy, *for example with Nginx*, then you need to leave these unset following upstream instructions

Reply:
I believe this may have more been of a recommendation *to* use Nginx over this, as most modules will do that over exposing TLS cert options like this and accessing services directly

I think it could be a good addition here as well, but not a blocker by any means

Reply:
I'm not exactly familiar with Nginx 🤦🏼‍♀️ Is there an example I can look at?

Reply:
[pixelfed](https://github.com/NixOS/nixpkgs/blob/ff5ffbd32324ee8b9f5caeafe904ab17688fcc17/nixos/modules/services/web-apps/pixelfed.nix#L96) uses this well. A minimal example would be

```nix
{
  options.myService = {
    # ...
    domain = lib.mkOption {
      type = lib.types.str;
      description = "FQDN for myService";
      example = "my-service.example.org";
    };

    nginx = lib.mkOption {
      type = lib.types.nullOr (
        lib.types.submodule (import ../web-servers/nginx/vhost-options.nix { inherit config lib; })
      );
      default = { };
      example = lib.literalExpression ''
        {
          enableACME = true;
          forceSSL = true;
        }
      '';
    };
  };

  config = lib.mkIf cfg.enable {
    # ...
    services.nginx = lib.mkIf (cfg.nginx != null) {
      enable = lib.mkDefault true;

      virtualHosts.${cfg.domain} = lib.mkMerge [
        {
          some = "nice";
          defaults = [
            "if"
            "any"
          ];
        }
        cfg.nginx
      ];
    };
  };
}
```